simulation/modules/afumigatus/__init__.py

```diff
@@ -0,0 +1,125 @@
+"""
+Afumigatus Module
+"""
+from math import floor
+
+import attr
+import numpy as np
+
+from simulation.module import Module, ModuleState
+from simulation.state import State
+from simulation.validation import ValidationError
+from simulation.modules.afumigatus.afumigatus_cell import AfumigatusCell
+
+from typing import List
+from typing import Dict
+
+
+@attr.s(kw_only=True, repr=False)
+class AfumigatusState(ModuleState):
+    #Afumigatus data struct
+    lastId:int = attr.ib(default=-1)
+    min_iter_to_status_change: int = attr.ib(default=0)
+    pr_status_change: float = attr.ib(default=0.5)
+	
+    #def __repr__(self):
+    #    return f'AfumigatusState(concentration, wind_x, wind_y, source)'
+    pass
+
+class Afumigatus(Module):
+    name = 'afumigatus'
+    defaults = {
+        'trees' : 'np.array([])',
+        'min_iter_to_status_change' : '5',
+        'pr_status_change' : '0.05',
+        'num_spore' : '0',
+        'lastId' : '-1'
+    }
+    
+    StateClass = AfumigatusState
+
+    def initialize(self, state: State):
+        afumigatus: AfumigatusState = state.afumigatus
+
+        c = self.config
+
+        afumigatus.trees = np.array([])
```

Comment:
This will have to go in the module state definition.  In particular, we will need to find a way to serialize it into an hdf5 file.

As a side note, I hadn't considered storing graphs, but I might be able to suggest a flattened internal representation (as a sparse matrix) that will be natural to save out to disk.  This might help us find ways to vectorize the computation to be more efficient as well.